core/src/main/scala/filodb.core/memstore/TimeSeriesMemStore.scala

```diff
@@ -25,6 +26,10 @@ extends MemStore with StrictLogging {
   type Shards = NonBlockingHashMapLong[TimeSeriesShard]
   private val datasets = new HashMap[DatasetRef, Shards]
 
+  // Memory pool for uncompressed ingestion data and partition keys, shared by all shards.
+  private val bufferMemoryManager = new NativeMemoryManager(
```

Comment:
Agreed. Solution is to have one memory manager per dataset

Reply:
Okay I guess. It seems like a mistake to me, because we'll have to tune memory settings to prevent wasting memory. Designs which favor tuning should be avoided as much as possible.